docs/appeng/adrs/0006-web-development-framework-preference.md

```diff
@@ -0,0 +1,130 @@
+
+(this list is likely not complete, feel free to modify or add to it as we put this draft together)
+
+The decision outcome does not need to be a single framework. Ideally we'll choose 2-3 that we would ask folks to choose from in most scenarios.
+
+## Decision
+
+Put a decision outcome here when ready.
+
+## Why is this applicable to the Practice as a whole?
+
+A large portion of our work is focused on developing web apps in partnership with public and private organizations. Each time something new starts we need to make decisions about how we'd like to approach it. This is meant to help give guidance in order to lower the burden of that choice.
+
+Having a 'practice opinion' on these tools will also help support folks' professional development, as well as our hiring and on-boarding processes. As we hire and get folks up to speed, this decision may be able to help us tailor what we do toward 'what we're trying to be good at.' Similarly we could lean on this as we seek to provide resources for folks developing their professional development plans.
+
+## Alternatives Considered
+
+### Django
+
+#### Pros
+
+- `+` Mature framework that's been around since 2005.
+- `+` Opinionated tooling for managing data migrations, potentially making the task more accessible for junior practitioners.
+- `+` Strong ORM allowing the abstraction of many database interactions.
+- `+` Features designed to make form creation and management easier.
+- `+` Strong templating tooling
+- `+` Possible to gain a portion of the features typically associated with single page applications, with less overhead by pairing the framework with something like [htmx](https://htmx.org/).
+- `+` Uses the python programming language which enjoys broad community support, and is useful across multiple subpractices like web development and data science.
+- `+` Provides batteries included tooling for authentication and authorization, caching, and an administrative console, among others. These make it easy to create a secure REST API quickly, making it a great option for rapid prototyping.
+- `+` Uses Python, which is considered easy to learn and is now frequently taught in both college and bootcamp programs.
+- `+` Strong community support
+- `+` Built-in Admin interface is useful for having a UI to interact with project database
+
+#### Cons
+
+- `-` Different languages between backend and frontend if Javascript is needed for client interactions.
+- `-` May need to use additional tooling for asynchronous workload use cases (message queues and workers).
+- `-` Python's dynamic typing may lead to more effort spent on unit tests. But this can be mitigated some through [type hinting](https://docs.python.org/3/library/typing.html)
+- `-` Since Django is highly opinionated, if you're new to it it'll seem like there's a lot of "magic" under the hood. Compared to smaller frameworks like Flask the learning curve is can be steep.
```

Comment:
**thought**: I think I actually see the opinionated side of it as a pro. A new person going into flask has to figure out how to even set up the database connection properly, where everything should live to keep the main app file from becoming too large and unwieldy, and make more decisions about what tools should be used for different aspects of it. Django makes those decisions and gives you easy escape hatches in the settings to change that if you need to.